### PR TITLE
[GPWS] Add minimums and 100above logic

### LINKED ISSUE #808: [BUG]Minimum Call
**Mod Version**
Master
**Describe the bug**
I set the mda value, compiled all the appr page, set the qnh value but  i haven't minimum call

**To Reproduce**

1.Set mda value
2.Starting approche
3.arrived at the minimum altitude there is no call

**Expected behavior**

**Actual behavior**

**References**

**Additional context**

Discord username (if different from GitHub):

### LINKED ISSUE #1461: [REQUEST] decision height/minimum callouts
**Is your feature request related to a problem? Please describe.**
No

**Describe the feature you'd like to see implemented**
Would like to see decision height and minimums call outs implemented

**References**

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by linkeleven**
Fixes #1461 [REQUEST] decision height/minimum callouts
Fixes #808 [BUG]Minimum Call 

## Summary of Changes

- Added logic for minimum and 100 above callout
- MDA will trigger the callouts based on indicated altitude (barometric altimeter)
- DH will trigger callouts based on altitude above ground (radio altimeter)

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build.py will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, slick on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by hotshotp** [comment]
I am adding the 100 above wwise event today when I get home I promise

**Comment by hotshotp** [comment]
added the event, it is merged

**Comment by hotshotp** [comment]
This works well, I tried both DH and MDA and worked for both. Thing is... with my new wind/fan/drag sounds all of the callouts are too quiet, so the 100 above (the new one) is good volume but the minimum (the asobo sound) is too quiet, so in my next PR i will add a louder minimums callout and hopefully a few others.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js [hunk 1]
```diff
@@ -25,18 +26,60 @@ class A32NX_GPWS {
...
+            this.gpws_dont_sink(altitude, vSpeed, phase);
+            this.gpws_too_low(altitude, Airspeed, FLAP_PushButton, phase);
         } else {
             SimVar.SetSimVarValue("L:GPWS_SINK_RATE", "Bool", 0);
             SimVar.SetSimVarValue("L:GPWS_PULL_UP", "Bool", 0);
             SimVar.SetSimVarValue("L:GPWS_DONT_SINK", "Bool", 0);
             SimVar.SetSimVarValue("L:GPWS_TOO_LOW", "Enum", 0);
         }
+
+        if ((mda != 0 || dh != 0) && phase == FlightPhase.FLIGHT_PHASE_APPROACH) {
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:45]:
```suggestion
        if ((mda !== 0 || dh !== 0) && phase === FlightPhase.FLIGHT_PHASE_APPROACH) {
```

**Comment by linkeleven** [comment]
converted to draft until PR #1564 and PR #1586 are merged

**Comment by linkeleven** [comment]
ready for code review and after that ready to test. current issue: altitude callouts are not inhibited when minimums or 100 above are playing. I will create an issue after merge because the altitude callouts need a rework.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js [hunk 2]
```diff
@@ -25,18 +26,60 @@ class A32NX_GPWS {
...
             SimVar.SetSimVarValue("L:GPWS_DONT_SINK", "Bool", 0);
             SimVar.SetSimVarValue("L:GPWS_TOO_LOW", "Enum", 0);
         }
+
+        if ((mda !== 0 || dh !== -1) && phase === FlightPhase.FLIGHT_PHASE_APPROACH) {
+            let minimumsDA; //MDA or DH
+            let minimumsIA; //radio or baro altitude
+            const radioAlt = SimVar.GetSimVarValue("PLANE ALT ABOVE GROUND MINUS CG", "Feet");
+            const baroAlt = SimVar.GetSimVarValue("INDICATED ALTITUDE", "feet");
+            if (dh > -1) {
```
Comment (paul92ilm) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:50]: `if(dh >= 0)` will be better no ? (-1 is the default value but negative DH is not allowed)

Comment (paul92ilm) [reply A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:50]: Else LGTM, good work

**Comment by jokey2k** [comment]
Name: JoKeY | Markus#0001
Date of testing: Current date 31/10/20
Version of the sim: 1.10.7
PR Tested: #1548
Object of test: Hundred above and minimum callouts

Testing technique:
Flown various arrivals from/to EDDB 25R with DH 540

Steps you've made to test the Pull Request:
Tried at least once from C&D as well as spawn on runway for takeoff

Flight notes
Tried following situations:
- Landing in one go
- Interrupting Autopilot to get no callout above RA 640
- Going from 1000 to 600 to 800 ft
- Going from 800 to 580 to 700 ft
- Going from 700 to 500 to 600 ft
- Landing

Startup: Notes on the startup (Everything is normal)
Taxi/Takeoff : Notes on the taxi an takeoff (Everything is normal)
Cruise: Notes on cruise (Everything is normal)
Descent/Landing : Notes on descent and landing (Work as expected)
Shutdown: none

Issues (If Any):
- Volume of callouts differ from RA height callouts
- callouts overlap with RA height callouts (listed as known issue, but verified)

Overall Rating: Good
Conclusions of the testing: Looks good

**Comment by lukecologne** [comment]
Removed the label since this branch is based on commit after the fix

**Comment by jokey2k** [comment]
Date: 02 Nov 2020
MSFS 1.10.8

Is broken when applying patch to current master branch. Battery shows 0.0V and the displays do not power on

**Comment by jokey2k** [comment]
Okay, ignore my comment, didn't notice I have to node-build now too. Works fine

**Comment by linkeleven** [comment]
thanks for testing!

**Comment by xdega** [comment]
Great work @linkeleven! Thanks for your efforts to implement this.

**Pull request merged**